sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_client.py

```diff
@@ -386,7 +395,9 @@ def get_subscription_receiver(self, topic_name, subscription_name, **kwargs):
                 retry_backoff_max=self._config.retry_backoff_max,
                 **kwargs
             )
-        else:
+        except ValueError:
+            if sub_queue: # If we got here and sub_queue is defined, it's an incorrect value or something unrelated.
+                raise
```

Comment:
I think it should be moved ahead of "handler = ServiceBusReceiver".
otherwise if it's a normal queue if any value raised by ServiceBusReceiver constructors would be omitted.

Reply:
(I see get_queue_receiver() you're checking ahead)

Reply:
never mind, your code is good to handle the case